buku.py

```diff
@@ -4294,7 +4314,12 @@ def main():
     search_opted = True
     update_search_results = False
 
-    if args.sany is not None:
+    if args.stag is not None and args.sany is not None:
```

Comment:
I think we can add a check for args.stags non None under args.sany and do this. Same goes for args.sall case and args.sreg case. As the priorities of sany, sall and sreg are greater than stag the code will never enter the `args.stag is not None` case in line 4341.

Reply:
We should also pull args.keywords above args.tags and treat it the same way as args.sany